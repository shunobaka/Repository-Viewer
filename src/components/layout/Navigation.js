/**
 * @fileoverview Defines a Navigation react component that renders a navbar.
 */
import React, { useState } from 'react';
import { Link, useHistory } from 'react-router-dom';
import { Navbar, Form, FormControl, Button } from 'react-bootstrap';

/**
 * Navigation react component that renders a navbar.
 */
const Navigation = () => {
  const [userInput, setUserInput] = useState('');
  const history = useHistory();

  const onChange = (e) => {
    setUserInput(e.target.value);
  };

  const onSubmit = (e) => {
    e.preventDefault();
    history.push(`/search/${userInput}`);
  };

  return (
    <Navbar bg="dark" expand="lg">
      <Navbar.Brand>
        <Link to="/" className="home-button">
          Home
        </Link>
      </Navbar.Brand>
      <Navbar.Toggle aria-controls="basic-navbar-nav" />
      <Navbar.Collapse id="basic-navbar-nav">
        <Form inline className="ml-auto" onSubmit={onSubmit}>
          <FormControl
            type="text"
            placeholder="Username"
            className="mr-sm-2"
            value={userInput}
            onChange={onChange}
          />
          <Link to={`/search/${userInput}`}>
            <Button variant="outline-success">Search</Button>
          </Link>
        </Form>
      </Navbar.Collapse>
    </Navbar>
  );
};

export default Navigation;
