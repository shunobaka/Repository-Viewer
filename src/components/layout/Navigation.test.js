/**
 * @fileoverview Defines snapshot unit tests for Navigation component.
 */
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navigation from './Navigation';

describe('navigation snapshots', () => {
  it('navigation renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Navigation />
        </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
