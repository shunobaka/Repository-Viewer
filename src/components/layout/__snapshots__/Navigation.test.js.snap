// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navigation snapshots navigation renders correctly 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-dark"
>
  <span
    className="navbar-brand"
  >
    <a
      className="home-button"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
  </span>
  <button
    aria-controls="basic-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="basic-navbar-nav"
  >
    <form
      className="ml-auto form-inline"
      onSubmit={[Function]}
    >
      <input
        className="mr-sm-2 form-control"
        onChange={[Function]}
        placeholder="Username"
        type="text"
        value=""
      />
      <a
        href="/search/"
        onClick={[Function]}
      >
        <button
          className="btn btn-outline-success"
          disabled={false}
          type="button"
        >
          Search
        </button>
      </a>
    </form>
  </div>
</nav>
`;
